hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/BucketManagerImpl.java

```diff
@@ -431,10 +432,11 @@ public boolean addAcl(OzoneObj obj, OzoneAcl acl) throws IOException {
             BUCKET_NOT_FOUND);
       }
 
-      changed = bucketInfo.addAcl(acl);
-      if (changed) {
-        metadataManager.getBucketTable().put(dbBucketKey, bucketInfo);
+      if (!bucketInfo.addAcl(acl)) {
+        throw new OMException("Add acl operation failed" + bucket,
+            ACL_ADD_FAILED);
```

Comment:
will this be an incompatible change, instead of returning false throw exception.
I thought you tend to change the ex caought at L440

Reply:
So @ayushtkn  think how to modify the incompatible changes is correct?Thank you.

Reply:
So, will returning false cause any issue?
We are catching exception bellow at L440, For that we can throw this ACL_ADD_FAILED exception. does that make sense?

Reply:
What is @ChenSammi 's opinion on this modification?Thank you.

Reply:
Ayush, IMHO strictly speaking, incompatible change implies public API method signature change. OMException is a subclass of IOException so it will be caught below.

That said, I don't think this is a refactor either. So maybe Ayush was misled.

Reply:
Couldn't check again, Wei-Chiu if you are convinced, Please proceed, nothing blocking from my side

Reply:
This seems to be a behavior change. Client previous expect false without exception for add existing acl or 
remove non-existing acl now need to handle the exception.